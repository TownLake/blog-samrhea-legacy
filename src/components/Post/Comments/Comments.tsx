import React from "react";

import { DiscussionEmbed } from "disqus-react";

import { useSiteMetadata } from "@/hooks";

interface Props {
  postTitle: string;
  postSlug: string;
}

const Comments: React.FC<Props> = ({ postTitle, postSlug }: Props) => {
  const { url, disqusShortname } = useSiteMetadata();

  if (!disqusShortname) {
    return null;
  }

  return (
    <DiscussionEmbed
      shortname={disqusShortname}
      config={{
        url: url + postSlug,
        identifier: postTitle,
        title: postTitle,
      }}
    />
  );
};

export default Comments;
