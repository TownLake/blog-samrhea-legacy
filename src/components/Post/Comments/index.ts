export { default as Comments } from "./Comments";
