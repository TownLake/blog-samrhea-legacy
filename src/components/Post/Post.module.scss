@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: center;
    text-align: center;
    @include margin(1, 0, 0, 0);

    .buttonArticles {
      @include margin(0, 0.5, 0, 0);
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .buttons {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin(0, 0.5, 0, 0);
    }
  }
}
