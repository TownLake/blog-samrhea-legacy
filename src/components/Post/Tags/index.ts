export { default as Tags } from "./Tags";
