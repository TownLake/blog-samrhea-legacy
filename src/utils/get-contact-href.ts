import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
