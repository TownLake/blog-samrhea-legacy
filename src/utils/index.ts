export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
