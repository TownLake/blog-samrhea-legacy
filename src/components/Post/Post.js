// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import styles from './Post.module.scss';
import type { Node } from '../../types';

type Props = {
  post: Node
};

const Post = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles['post']}>
      <Link className={styles['post__home-button']} to="/">Home</Link>

      <div className={styles['post__content']}>
        <Content body={html} title={title} />
      </div>

      <div className={styles['post__footer']}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles['post__comments']}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
