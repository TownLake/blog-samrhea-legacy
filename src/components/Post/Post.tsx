import React from "react";

import { Button } from "@/components/Button";
import { ThemeSwitcher } from "@/components/ThemeSwitcher";
import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

interface Props {
  post: Node;
}

const Post: React.FC<Props> = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <div className={styles.buttons}>
        <Button className={styles.buttonArticles} title="All Articles" to="/" />
        <ThemeSwitcher />
      </div>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
