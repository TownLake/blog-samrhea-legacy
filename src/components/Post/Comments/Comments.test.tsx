import React from "react";

import { StaticQuery, useStaticQuery } from "gatsby";

import { Comments } from "@/components/Post/Comments";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Comments", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );

    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const props = {
      postTitle: mocks.markdownRemark.frontmatter.title,
      postSlug: mocks.markdownRemark.fields.slug,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Comments {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
