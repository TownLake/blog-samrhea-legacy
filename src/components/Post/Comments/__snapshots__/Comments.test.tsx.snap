// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders correctly 1`] = `null`;
